batcher/aligned-batcher/src/lib.rs

```diff
@@ -51,21 +51,43 @@ const ADDITIONAL_SUBMISSION_COST_PER_PROOF: u128 = 13_000;
 const CONSTANT_COST: u128 = AGGREGATOR_COST + BATCHER_SUBMISSION_BASE_COST;
 const MIN_BALANCE_PER_PROOF: u128 = ADDITIONAL_SUBMISSION_COST_PER_PROOF * 100_000_000_000; // 100 Gwei = 0.0000001 ether (high gas price)
 
+struct BatchState {
+    batch_queue: BatchQueue,
+    user_nonces: HashMap<Address, U256>,
+    user_proof_count_in_batch: HashMap<Address, u64>,
+}
+
+impl BatchState {
+    fn new() -> Self {
+        Self {
+            batch_queue: BatchQueue::new(),
+            user_nonces: HashMap::new(),
+            user_proof_count_in_batch: HashMap::new(),
+        }
+    }
+
+    fn increment_user_proof_count(&mut self, addr: &Address) -> u64 {
+        self.user_proof_count_in_batch.get(addr).unwrap_or(&0) + 1
+    }
+
+    fn update_user_proof_count(&mut self, addr: &Address, count: u64) {
+        self.user_proof_count_in_batch.insert(*addr, count);
+    }
+}
```

Comment:
we always update by 1, right? if so, this function could actually be `increment_user_proof_count`. Check the `entry()` [API for hashmaps](https://doc.rust-lang.org/std/collections/struct.HashMap.html#method.entry).x
This method should be called after `check_user_balance` returns true, or the function should be renamed to `check_user_balance_and_increment_proof_count`

Reply:
Solved in [5e72f95](https://github.com/yetanotherco/aligned_layer/pull/679/commits/5e72f951b9886bef437e6f333b5c8db2b8613ec8)